### DISCUSSION TITLE
Use of generic Field type is misleading

### DISCUSSION THREAD

**Discussion by bdoubrov** [pdf-association/arlington-pdf-model#56] [General] [answered]
Currently **Field** type does not define **FT** entry, meaning that it is probably used as a non-terminal node in the fields tree. However, it is also commonly used a type of a link for **P** (parent) keys in annotations and terminal fields. 

As currently Arlington model is not really object-oriented, there is no mechanism to say that a link to a **Field** has to be replaced by a one of the strongly typed **Field** types such as  **FieldChoice** or others.

I would suggest specifying all possible field types instead of just one **Field** as it is now, even if this might become less convenient.

One particular example is **P** key in **AnnotWidget**, but there are other places in the model where **P** key specifies only **Field** link.

**Comment by bdoubrov** [comment]
This might be a duplicate or at least related to https://github.com/pdf-association/arlington-pdf-model/issues/28

**Comment by petervwyatt** [comment, marked as answer]
Yes, it is the same as #28.
